File: src/main/java/com/borvalcode/sudoku/infrastructure/delivery/armeria/controller/JsonGet.java
package com.borvalcode.sudoku.infrastructure.delivery.armeria.controller;

import com.borvalcode.sudoku.infrastructure.delivery.armeria.JsonConverter;
import com.linecorp.armeria.server.annotation.ProducesJson;
import com.linecorp.armeria.server.annotation.ResponseConverter;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
@ResponseConverter(JsonConverter.class)
@ProducesJson
public @interface JsonGet {
}

File: src/main/kotlin/com/borvalcode/sudoku/infrastructure/delivery/armeria/controller/JsonGet.kt
package com.borvalcode.sudoku.infrastructure.delivery.armeria.controller

import com.borvalcode.sudoku.infrastructure.delivery.armeria.JsonConverter
import com.linecorp.armeria.server.annotation.ProducesJson
import com.linecorp.armeria.server.annotation.ResponseConverter

@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@ResponseConverter(JsonConverter::class)
@ProducesJson
annotation class JsonGet 